R/MultilabelDBRWrapper.R

```diff
@@ -0,0 +1,95 @@
+#' @title Use dependent binary relevance method (DBR) to create a multilabel learner.
+#'
+#' @description
+#' Every learner which is implemented in mlr and which supports binary
+#' classification can be converted to a wrapped DBR multilabel learner.
+#' The multilabel classification problem is converted into simple binary classifications
+#' for each label/target on which the binary learner is applied. 
+#' For each target, actual information of all binary labels (except the target variable) is used as additional features.
+#' At prediction time these labels need to be predicted by the level-1 learner.
+#'
+#' Models can easily be accessed via \code{\link{getLearnerModel}}.
+#'
+#' @param level1Learner The learner which produces predicted labels at prediction time. Default is \code{learner}, i.e. the same learner as the passed one. 
+#' @template arg_learner
+#' @template ret_learner
+#' @references
+#' Montanes, E. et al. (2013)
+#' \emph{Dependent binary relevance models for multi-label classification}
+#' Artificial Intelligence Center, University of Oviedo at Gijon, Spain.
+#' @family wrapper
+#' @family multilabel
+#' @export
+#' @examples
+#' d = getTaskData(yeast.task)
+#' # drop some labels so example runs faster
+#' d = d[, c(1:3, 15:117)]
+#' task = makeMultilabelTask(data = d, target = c("label1", "label2", "label3"))
+#' lrn = makeMultilabelDBRWrapper("classif.rpart", level1Learner = "classif.rpart")
+#' lrn = setPredictType(lrn, "prob")
+#' # train, predict and evaluate
+#' mod = train(lrn, task)
+#' pred = predict(mod, task)
+#' p = performance(pred)
+#' performance(pred, measure = hamloss)
+#' getMultilabelBinaryPerformances(pred, measures = list(mmce, auc))
+#' # above works also with predictions from resample!
+makeMultilabelDBRWrapper = function(learner, level1Learner = learner) {
```

Comment:
Same as in stacking, do we need two learners? and maybe use MultilabelBinaryRelevanceWrapper?